### ISSUE TITLE
Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED

### ISSUE THREAD

**Issue by jinfagang** [tensorflow/tensorflow#22056]
TensorFlow: 1.10.1
cuda: 9.0
cudnn: cudnn7.1 for cuda9.0
python3
Ubuntu 18.04
nvidia-driver: 390

Try running cnns error got:

```
Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2018-09-04 23:13:52.571542: E tensorflow/stream_executor/cuda/cuda_dnn.cc:360] Possibly insufficient driver version: 390.77.0

```
I think the cuda version are already the newest version, and my cudnn and cuda are OK. Why still got this problem? The error message throw from cuda_dnn.cc should be more specific what's going on.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by byronyi** [comment]
Same problem here. Log:

```
2018-09-27 10:02:39.079409: E tensorflow/stream_executor/cuda/cuda_dnn.cc:352] Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2018-09-27 10:02:39.079563: E tensorflow/stream_executor/cuda/cuda_dnn.cc:360] Possibly insufficient driver version: 396.26.0
```

Have I written custom code: Yes
OS Platform and Distribution: COS on GCP (Host), Ubuntu 18.04 (Docker)
TensorFlow installed from: pip
TensorFlow version: 1.10.1
Bazel version: N/A
CUDA/cuDNN version: 9.0/7.2
GPU model and memory: K80c, 12GiB
Exact command to reproduce: Used in a COCO retrained task
Mobile device: N/A

**Comment by byronyi** [comment]
I found my problem is introduced by installing the wrong version of CUDNN. @jinfagang could you try `nvidia-smi` and `dpkg -l | grep cudnn` and paste the output here?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by saskra** [comment]
> Is it really the driver version being the problem?

No, the problem was an incompatible Cudnn version (7.1.2) being installed by creating an Anaconda environment which I had to manually overwrite with the previously installed system's version (7.3.1).

**Comment by arham-saeed** [comment]
> > Is it really the driver version being the problem?
> 
> No, the problem was an incompatible Cudnn version (7.1.2) being installed by creating an Anaconda environment which I had to manually overwrite with the previously installed system's version (7.3.1).

I'm having the same problem and I'm also working on a conda environment. Can you tell me how to overwrite the previous version?

**Comment by saskra** [comment]
Actually, I finally succeeded in installing the right driver version and do not have to overwrite cuDNN anymore: https://github.com/pplcc/ubuntu-tensorflow-pytorch-setup/issues/1

**Comment by ymodak** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by AndreaIncertiDelmonte** [comment]
In my case apt updated:
- libcudnn7 from 7.2.1.38-1+cuda9.0 to 7.4.2.24-1+cuda10.0
- libnccl2 from 2.2.13-1+cuda9.0 to 2.3.7-1+cuda10.0
causing the same issue in TF.
A way to prevent that could be `sudo apt-get purge nvidia-machine-learning-repo-ubuntu1604`

**Comment by lemr** [comment]
I had the same/similar problem:
```2019-02-20 15:39:47.506767: E tensorflow/stream_executor/cuda/cuda_dnn.cc:373] Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2019-02-20 15:39:47.506890: E tensorflow/stream_executor/cuda/cuda_dnn.cc:381] Possibly insufficient driver version: 390.77.0
2019-02-20 15:39:47.506951: W tensorflow/core/framework/op_kernel.cc:1273] OP_REQUIRES failed at cudnn_rnn_ops.cc:1214 : Unknown: Fail to find the dnn implementation.
```

using docker container with dockerfile:
```FROM tensorflow/tensorflow:1.12.0-gpu-py3
...
RUN apt-get update -y && apt-get install -y python3-dev python3-pip python3-lxml
...```
with Keras layer `CuDNNLSTM`.

I believe the problem was installing `python3-dev python3-pip python3-lxml`. When I removed that line the error disappeared.

**Comment by soylisK** [comment]
I get the same problem after updating, before update everything was running with no problem
could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
possibly insufficient driver version: 410.104.0

I am using ubuntu-18.04LTS, python3.6.7, tensorflow 1.9, I've install it by using bazel -0.17.2 build from source, cuda 10.0, cudnn-7.5.0 and my gpu is titan xp

Any ideas? Maybe rollback to an older version of the driver?

**Comment by bzamecnik** [comment]
Yeah, in my case I ran TF 1.13  using CUDA 9.0 with CuDNN for CUDA 10.0 (since I was testing new TF 2.0). Downgrade `sudo apt install libcudnn7=7.5.0.56-1+cuda9.0` solved it.

**Comment by soylisK** [comment]
I found this, after all, to work for me :
 

> -from  software & updates > Additional Drivers > choose nvidia-418(downloads and install it) 
> - reboot PC

As result got upgraded to cuda-10.1 (from 10.0). It works for now !

**Comment by purneshdasari** [comment]
For windows 10 .
TF - 1.12.0,
Cuda - 9
Cudnn - 7.4.2

Close the jupyter notebook and restart it . It will work fine.
but when we open another jupyter notebook involving cuda , it will again cause issues.
so closing jupyter and working on one notebook at a time is the solution for me.
